.circleci/config.yml

```diff
@@ -19,58 +20,48 @@ push_builds_only: &push_builds_only
 executors:
   amphtml-docker-small-executor:
     docker:
-      - image: cimg/base:2021.05
+      - image: cimg/base:stable
     resource_class: small
   amphtml-medium-executor:
-    machine:
-      image: ubuntu-2004:202010-01
+    docker:
+      - image: cimg/node:lts-browsers
     resource_class: medium
   amphtml-large-executor:
-    machine:
-      image: ubuntu-2004:202010-01
+    docker:
+      - image: cimg/node:lts-browsers
     resource_class: large
   amphtml-xlarge-executor:
-    machine:
-      image: ubuntu-2004:202010-01
+    docker:
+      - image: cimg/node:lts-browsers
+    resource_class: xlarge
+  validator-xlarge-executor:
+    docker:
+      - image: cimg/openjdk:16.0-node
     resource_class: xlarge
 
 commands:
-  restore_workspace:
+  setup_vm:
     steps:
       - attach_workspace:
           at: /tmp
       - run:
-          name: 'Rename Restored Workspace'
+          name: 'Rename Restored Workspace and Create New Workspace'
           command: |
-            if [[ -d /tmp/workspace ]]; then
-              mv /tmp/workspace /tmp/restored-workspace
-            fi
-  maybe_gracefully_halt:
-    steps:
+            mv /tmp/workspace /tmp/restored-workspace
+            mkdir -p /tmp/workspace
       - run:
           name: 'Maybe Gracefully Halt'
           command: curl -sS https://raw.githubusercontent.com/ampproject/amphtml/main/.circleci/maybe_gracefully_halt.sh | bash
-  setup_vm:
-    steps:
-      - run:
-          name: 'Create Workspace'
-          command: mkdir -p /tmp/workspace
       - checkout
       - run:
-          name: 'Fetch Merge Commit'
-          command: ./.circleci/fetch_merge_commit.sh
-      - run:
-          name: 'Check Config'
-          command: ./.circleci/check_config.sh
-      - run:
-          name: 'Configure Hosts'
-          command: cat ./build-system/test-configs/hosts | sudo tee -a /etc/hosts
-      - run:
-          name: 'Install Dependencies'
-          command: ./.circleci/install_dependencies.sh
-      - run:
-          name: 'Restore Build Output'
-          command: ./.circleci/restore_build_output.sh
+          name: 'Fetch Merge Commit, Check Config, Restore Build Output, and Configure Hosts'
```

Comment:
I love the idea of consolidating these into one step. Perhaps shorten the name to something like "Setup Environment"?

Reply:
We have too many steps with the title "setup" in them... I'm open to other suggestions :D

Reply:
Fair! How about "Configure environment"?

Reply:
Went with `Configure Development Environment`